### PR TITLE
feat: number convertor

### PR DESCRIPTION
**Pull request by Daooobg**
This is what i did

### CODE REVIEW

File: 04-dom/exercise/09-numberConvertor.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+function solve() {
+
+    let optionBinary = document.createElement("OPTION");
```
Comment (antoniaat) [inline 04-dom/exercise/09-numberConvertor.js]: Why not using `const` here as well?

Comment (antoniaat) [reply 04-dom/exercise/09-numberConvertor.js]: +When we are creating or taking an element reference I suggest to always put name in the end so to remember that inside we have live DOM element.

File: 04-dom/exercise/09-numberConvertor.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    const result = document.getElementById('result');
+
+    optionBinary.textContent = 'Binary';
+    optionBinary.value = 'binary';
+    optionHexadecimal.textContent = 'Hexadecimal';
+    optionHexadecimal.value = 'hexadecimal';
+    selectMenuTo.appendChild(optionBinary);
+    selectMenuTo.appendChild(optionHexadecimal);
+
+    btn[0].onclick = function () {
```
Comment (antoniaat) [inline 04-dom/exercise/09-numberConvertor.js]: What is btn[0]? I suggest creating a variable with descriptive name for this.

File: 04-dom/exercise/09-numberConvertor.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        let result = [];
+        while (num > 0) {
+            let digit = num % 2;
+            result.unshift(digit);
+            num = Math.floor(num / 2);
+        }
+        return result.join('');
+    }
+
+    function convertToHeXadecimal(num) {
```
Comment (antoniaat) [inline 04-dom/exercise/09-numberConvertor.js]: Of course this function works fine but if you have the wish you can research for some ways to improve it. I would suggest something like this (not tested, just a suggestion):

```javascript
function convertToHexadecimal(num) {
  // Base case: if num is 0, return '0' (special case)
  if (num === 0) return '0';

  // Define a string containing all hexadecimal characters in order
  const hexChars = '0123456789ABCDEF';

  let result = '';

  // Loop until num becomes 0
  while (num > 0) {
    // Calculate the remainder when dividing by 16 (hexadecimal base)
    const digit = num % 16;

    // Prepend the corresponding hexadecimal character to the result
    result = hexChars[digit] + result;

    // Reduce num by the integer division by 16
    num = Math.floor(num / 16);
  }

  return result;
}
```

**Review by antoniaat — CHANGES_REQUESTED** [review]
The branch should point to main - please rebase.

**Review by antoniaat — APPROVED** [review]
Looks much better! Great job!

**Review by zlatozaraZlatkova — APPROVED** [review]

**Pull request merged**